generators/client/templates/angular/src/main/webapp/app/entities/user/user.model.ts.ejs

```diff
@@ -20,8 +20,8 @@
   const idType = getTypescriptKeyType(user.primaryKey.type);
 _%>
 export interface IUser {
-  id: <%= idType %>;
-  login: string;
+  id?: <%= idType %>;
+  login?: string;
```

Comment:
Why this change? When can `id` or `login` can be undefined?

Reply:
`login` can be undefined using dto:
https://github.com/jhipster/generator-jhipster/blob/ac3f14e815db7e419b4f67541d4c00573aed2bdc/generators/server/templates/src/main/java/package/service/mapper/UserMapper.java.ejs#L142-L152

Not sure about `id`, but I am wondering if it's worth to keep it as non-undefined and keep the quirk only for user entity..

Reply:
OK, good point about keeping `User` entity similar to other entities, so this is good to go.